docs/training/apis-howto/api-core-ug.rst

```diff
@@ -1,212 +1,302 @@
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :start-at: logging.basicConfig
-      :end-at: logging.error
+In the WebUI, select your experiment, and then navigate to the **Logs** tab.
 
-#. In your ``if __name__ == "__main__"`` block, wrap the entire execution of ``main()`` within the
-   scope of :meth:`determined.core.init`, which prepares resources for training and cleans them up
-   afterward. Add the ``core_context`` as a new argument to ``main()`` because the Core API is
-   accessed through the ``core_context`` object.
+************************
+ Step 2: Report Metrics
+************************
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :start-at: with det.core.init
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a :class:`~determined.core.Context` object to allow
+interaction with the master. Then, we'll pass the ``core_context`` as an argument into ``main()``,
+``train()``, and ``test()`` and modify the function headers accordingly.
 
-#. Within ``main()``, add two calls: (1) report training metrics periodically during training and
-   (2) report validation metrics every time a validation runs.
+To run our experiment, we'll use the ``model_def_metrics.py`` script and its accompanying
+``metrics.yaml`` experiment configuration file.
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :pyobject: main
+Begin by importing Determined:
 
-   The ``report_validation_metrics()`` call typically happens after the validation step, however,
-   actual validation is not demonstrated by this example.
+.. code:: python
 
-#. Create a ``1_metrics.yaml`` file with an ``entrypoint`` invoking the new ``1_metrics.py`` file.
-   You can copy the ``0_start.yaml`` configuration file and change the first couple of lines:
+   import determined as det
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.yaml
-      :language: yaml
-      :lines: 1-2
+Step 2.1: Modify the Main Loop
+==============================
 
-#. Run the code using the command:
+We'll need a ``core.Context`` object for interacting with the master. To accomplish this, we'll
+modify the __main__loop to include ``core_context``:
 
-   .. code:: bash
+.. note::
 
-      det e create 1_metrics.yaml . -f
+   Refer to the ``if __name__ == "__main__":`` block in ``model_def_metrics.py``
 
-#. You can now navigate to the new experiment in the WebUI and view the plot populated with the
-   training and validation metrics.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start after: modify main loop core context
+   :end-at: main(core_context=core_context)
 
```

Comment:
you are missing a reference to your `include core context` snippet

Reply:
thanks! resolved